examples/inference/unified.py

```diff
@@ -0,0 +1,215 @@
+        latents: Optional[torch.FloatTensor] = None,
+        output_type: Optional[str] = "pil",
+    ):
+
+        if isinstance(prompt, str):
+            batch_size = 1
+        elif isinstance(prompt, list):
+            batch_size = len(prompt)
+        else:
+            raise ValueError(f"`prompt` has to be of type `str` or `list` but is {type(prompt)}")
+
+        if strength < 0 or strength > 1:
+            raise ValueError(f"The value of strength should in [0.0, 1.0] but is {strength}")
+
+        # set timesteps
+        accepts_offset = "offset" in set(inspect.signature(self.scheduler.set_timesteps).parameters.keys())
+        extra_set_kwargs = {}
+        offset = 0
+        if accepts_offset:
+            offset = 1
+            extra_set_kwargs["offset"] = 1
+
+        self.scheduler.set_timesteps(num_inference_steps, **extra_set_kwargs)
+
+        # get the original timestep using init_timestep
+        init_timestep = int(num_inference_steps * strength) + offset
+        init_timestep = min(init_timestep, num_inference_steps)
+        timesteps = self.scheduler.timesteps[-init_timestep]
+        timesteps = torch.tensor([timesteps] * batch_size, dtype=torch.long, device=self.device)
+
+        if init_image is None:
+            # get the initial random noise unless the user supplied it
+            latents_shape = (batch_size, self.unet.in_channels, height // 8, width // 8)
+            if latents is None:
+                latents = torch.randn(
+                    latents_shape,
+                    generator=generator,
+                    device=self.device,
+                )
+            else:
+                if latents.shape != latents_shape:
+                    raise ValueError(f"Unexpected latents shape, got {latents.shape}, expected {latents_shape}")
+                latents = latents.to(self.device)
+            init_latents=latents
+        else:
+            # preprocess image
+            init_image = preprocess(init_image).to(self.device)
+
+            # encode the init image into latents and scale the latents
+            init_latents = self.vae.encode(init_image).sample()
+            init_latents = 0.18215 * init_latents
+            init_latents_orig = init_latents
+
+            # prepare init_latents noise to latents
+            init_latents = torch.cat([init_latents] * batch_size)
+
+            # add noise to latents using the timesteps
+            noise = torch.randn(init_latents.shape, generator=generator, device=self.device)
+            init_latents = self.scheduler.add_noise(init_latents, noise, timesteps)
+
+        if mask_image is not None:
+            # preprocess mask
+            mask = preprocess_mask(mask_image).to(self.device)
+            mask = torch.cat([mask] * batch_size)
+            # check sizes
+            if not mask.shape == init_latents.shape:
+                raise ValueError(f"The mask and init_image should be the same size!")        
```

Comment:
flake8 reports: `F541 f-string is missing placeholders`, the `f` is not necessary